_tools/generate_release_cycle.py

```diff
@@ -70,38 +56,84 @@ def write_csv(self) -> None:
-            versions_by_category[cat][version] = row
+            versions_by_category[cat][details['key']] = row
 
         for cat, versions in versions_by_category.items():
             with open(f"include/{cat}.csv", "w", encoding="UTF-8", newline="") as file:
                 csv_file = csv.DictWriter(file, fieldnames=headers, lineterminator="\n")
                 csv_file.writeheader()
                 csv_file.writerows(versions.values())
 
-    def write_mermaid(self) -> None:
-        """Output Mermaid file."""
-        out = [MERMAID_HEADER]
+    def write_svg(self) -> None:
+        """Output SVG file."""
+        env = jinja2.Environment(
+            loader=jinja2.FileSystemLoader('_tools/'),
+            autoescape=True,
+            undefined=jinja2.StrictUndefined,
+        )
+        template = env.get_template("release_cycle_template.svg")
+
+        # Scale. Should be roughly the pixel size of the font.
+        # All later sizes are miltiplied by this, so you can think of all other
+        # numbers being multiples of the font size, like using `em` units in
+        # CSS.
+        # (Ideally we'd actually use `em` units, but SVG viewBox doesn't take
+        # those.)
+        SCALE = 18
+
+        # Width of the drawing and main parts
+        DIAGRAM_WIDTH = 46
+        LEGEND_WIDTH = 7
+        RIGHT_MARGIN = 0.5
+
+        # Height of one line. If you change this you'll need to tweak
+        # some positioning nombers in the template as well.
+        LINE_HEIGHT = 1.5
+
+        first_date = min(
+            ver['first_release_date'] for ver in self.sorted_versions
+        )
+        last_date = max(
+            ver['end_of_life_date'] for ver in self.sorted_versions
+        )
+
+        def date_to_x(date):
+            """Convert datetime.date to a SVG X coordinate"""
+            num_days = (date - first_date).days
+            total_days = (last_date - first_date).days
+            ratio = num_days / total_days
+            x = ratio * (DIAGRAM_WIDTH - LEGEND_WIDTH - RIGHT_MARGIN)
+            return x + LEGEND_WIDTH
+
+        def year_to_x(year):
+            """Convert year number to a SVG X coordinate of 1st January"""
+            return date_to_x(dt.date(year, 1, 1))
 
-        for version, details in reversed(self.versions.items()):
-            v = MERMAID_SECTION.format(
-                version=version,
-                first_release=details["first_release"],
-                eol=mermaid_date(details["end_of_life"]),
-                release_status=details["status"],
-                mermaid_status=MERMAID_STATUS_MAPPING[details["status"]],
-            )
-            out.append(v)
+        def format_year(year):
+            """Format year number for display"""
+            return f"'{year % 100:02}"
 
         with open(
-            "include/release-cycle.mmd", "w", encoding="UTF-8", newline="\n"
+            "include/release-cycle.svg", "w", encoding="UTF-8",
```

Comment:
```suggestion
            "include/release-cycle.svg", "w", encoding="UTF-8", newline="\n"
```

Why not keep this? Without it, as discussed on the previous PR, this checked in file will be written with inconsistent line endings on different platforms, which can lead to serious annoyances with Git.